docs/canonicalk8s/snap/explanation/security/auth.md

```diff
@@ -0,0 +1,116 @@
+
+The Kubernetes API server can be configured to accept OpenID Connect (OIDC)
+tokens for authentication from external identity providers.
+
+In {{product}}, anonymous API access is disabled by default.
+
+More details: [Kubernetes Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/)
+
+## Authorization
+
+After authentication, Kubernetes checks whether the user or service account is
+authorized to perform a requested action. In {{product}}, this is done through
+Role-Based Access Control.
+
+#### Role-Based Access Control (RBAC)
+
+RBAC Authorization in {{product}} is done through the following types of
+Kubernetes objects:
+
+- **Roles** define a set of _allow_ permissions within a namespace.
+- **ClusterRoles** define cluster-wide _allow_ permissions.
+- **RoleBindings** and **ClusterRoleBindings** assign these roles to users or
+  service accounts.
+
+Example use cases:
+- Granting users or service accounts read-only access to logs in a namespace.
+- Granting read-only access to nodes, services, and pods to a monitoring operator.
+
+Kubernetes defines a set of `ClusteRoles` and `ClusterRoleBindings` that apply
+to all users and service accounts, including the `default` service accounts
+mounted in pods:
+
+- `system:basic-user`: Allows a user read-only access to basic information about themselves.
+- `system:discovery`: Allows read-only access to API discovery endpoints needed
+  to discover and negotiate an API level.
+- `system:public-info-viewer`: Allows read-only access to non-sensitive
+  information about the cluster.
+
+More details:
+- [Kubernetes RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
+- [Default roles and role bindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings)
+
+## Admission controllers
+
+After a request has been authenticated and authorized, admission controllers
+are used to **validate** and / or **mutate** requests to the Kubernetes API that
+create, modify, or delete resources in Kubernetes. Admission controllers cannot
+block **get**, **list**, or **watch** requests.
+
+Kubernetes has a default list of admission controllers, which can be expanded
+or contracted. Keep in mind that several important features of Kubernetes
+require admission controller to be enabled in order to properly support them.
+As a result, a Kubernetes API server that is not properly configured with the
+right set of admission controllers is an incomplete server and will not support
+all the features you expect.
+
+In addition to the default list of admission controllers, {{product}} enables
+the `NodeRestriction` admission controller. This controller limits the `Node`
+and `Pod` objects a kubelet can modify, allowing them to modify only their
+own `Node` object and `Pod` objects bound to their node.
+
+More details: [Kubernetes Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/)
```

Comment:
Nit: For more details see [Kubernetes Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) upstream docs.

Reply:
Done